### ISSUE TITLE
Disable certain 'data:' URLs types in address bar

### ISSUE THREAD

**Issue by garvankeeley** [mozilla-mobile/firefox-ios#6315]
https://bugzilla.mozilla.org/show_bug.cgi?id=1624167

Certain data URL types should not be allowed, see https://blog.mozilla.org/security/2017/11/27/blocking-top-level-navigations-data-urls-firefox-59/

The important bit from that link is to allow the following:
``` the following cases will be allowed:
    1) User explicitly entering/pasting “data:…” into the address bar
    2) Opening all plain text data files
    3) Opening “data:image/*” in top-level window, unless it’s “data:image/svg+xml”
    4) Opening “data:application/pdf” and “data:application/json”
    5) Downloading a data: URL, e.g. ‘save-link-as’ of “data:…”
```

To avoid security risks we are going to only allow 2,3, and 4. The others are too risky IMO. 
That said, if there is interest in attempting 1 and 5 in a secure manner, please ping me before attempting this, not to dissuade you from it, but there might be additional consideration to discuss.

The test page for this is https://www-archive.mozilla.org/quality/networking/testing/datatests.html

**Comment by SimonBasca** [comment]
Verified as fixed on 25.0 (17793). Only 2,3, and 4 are allowed.

**Comment by Rukario** [comment]
Oh, this is it, you've killed the data url scheme I use. I am on 1 in this list which was allowed by official Firefox but seems not on iOS because of your "too risky IMO". Too risky how? Please, there's no need to create discrepancy between Firefox and Firefox for iOS.
